scripts/assemble.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+
+"""
+Generate the monolithic doctest.h header file.
+
+Our original script, scripts/cmake/assemble_single_header.cmake,
+simply read doctest_fwd.h and doctest.cpp and pasted them in-order,
+with some defines sandwiched inbetween.
+
+This script attempts to be a bit more future-proof, by automatically
+scanning for any .h and .cpp files that could be included,
+(topologically) sorting them by #include order, then creating
+the final header file.
+"""
+
+# /// script
+# requires-python = ">=3.6"
+# ///
+
+
+from __future__ import annotations
+
+import re
+import sys
+from pathlib import Path
+from typing import Generator, NoReturn
+
+
+def main(args: list[str]) -> NoReturn:
+  """Script entry-point."""
+
+  if len(args) != 1:
+    print("Usage: scripts/assemble.py", file=sys.stderr)
+    sys.exit(1)
+
+  script = Path(__file__).resolve()
+  root   = script.parent.parent
+  incdir = root / "doctest" / "parts"
```

Comment:
```suggestion
  src_dir = root / "doctest" / "parts"
```

As we are also reading `.cpp` files.